### ISSUE TITLE
About pre-tran and zero-shot

### ISSUE THREAD

**Issue by iro-del** [ls1rius/WSI_FiVE#19]
Hi，
Thank you for sharing your excellent work.
May I ask if you are utilizing DISMIL provided features or DISMIL provided wsi images in the model pre-training? If the features provided by DISMIL are utilized does it mean that there is no need to train the image encoder during the pre-training process? In addition, after pre-training to get the PTH according to the guidance in the TRIAN section of the readme, how should I zeroshot to get the corresponding results in the paper?

Looking forward to your reply！Thanks

**Comment by ls1rius** [comment]
This paper provides two solutions, one is to directly use the features provided by DSMIL for training, in which case there is no need to train the image encoder; Another approach is end2end training, in which case the images provided by DSMIL should be used. You can switch modes through IS_IMG_PTH config.
The experimental results of this article comprehensively consider both solutions.
To save GPU, you can use the default setting, which is to use DSMIL pre-trained PTH.

After pre-training, you can obtain zero shot results by encoding the description of pathology in text and calculating its similarity with image embeddings, similar to CLIP. You can refer to the code or the file './WSI_FiVE/data/TCGA-lung-patch_report/LUAD_LUSC_labels_val_reid.csv'

**Comment by iro-del** [comment]
Thanks for your reply! Another thing that confuses me is that during the few-shot learning , is the training dataset used the tcga and the corresponding text report taken from the pre-training dataset? I didn't find  relevant information about few-shot learning in  readme.

**Comment by ls1rius** [comment]
few shot file: [luad_tcga_pub_clinical_data.tsv](https://github.com/ls1rius/WSI_FiVE/blob/main/gpt_preprocess/luad_tcga_pub_clinical_data.tsv)

**Comment by ls1rius** [comment]
You can randomly sample n samples from the training set for training, which is the n-shot.

Normally, model is fine-tuned using labeled data based on pre-trained parameters.
However, if you only use the n-shot dataset (image text pairs) for pre training, and then use the labels of this dataset for fine-tuning, you can also achieve the results reported in our paper.

If you want to skip pre training and directly perform image-label training for few-shot, we haven't tried it yet. You can give it a try, and we look forward to seeing the results.

**Comment by iro-del** [comment]
Thanks for your patience! I think I got it. the train in the readme is used for pre-training, which can be done with the features provided by dismil, or the image encoder can be trained using WSI during pre-training. In the few-shot and fully supervised experiments, the images and corresponding labels are used to fine-tune on top of the pre-training. Please correct me if I'm not understanding  correctly.
but，which file should I utilize if I want to perform zero-sample classification of LUAD subtypes? What are the training and test sets used in the few-shot and fully supervised experiments? How should I obtain it? Looking forward to your reply. Thanks a lot

**Comment by ls1rius** [comment]
correct.

For zero-shot classification of LUAD subtypes, use [luad_tcga_pub_clinical_data.tsv](https://github.com/ls1rius/WSI_FiVE/tree/main/gpt_preprocess/luad_tcga_pub_clinical_data.tsv).
We only perform the zero-shot on the subtype dataset. And end2end training is recommended on this dataset.

few-shot and fully supervised experiments: follow [fix_pth.yaml](https://github.com/ls1rius/WSI_FiVE/tree/main/configs/wsi/fix_pth.yaml) and [fix_pth_ft.yaml](https://github.com/ls1rius/WSI_FiVE/tree/main/configs/wsi/fix_pth_ft.yaml)

**Comment by ls1rius** [comment]
In fact, models trained in this way have relatively strong performance.
Even no need to fine-tune.

**Comment by iro-del** [comment]
Thanks for your reply! but  I don't quite understand how to follow fix_pth.yaml and fix_pth_ft.yaml to perform a fewshot and fully supervised experiments, can you describe it in more detail?And, isn't fix_pth.yaml used for pre-training or am I misunderstanding?In fix_pth_ft.yaml, LUAD_LUSC_data_train_reid.csv stores the training fully-supervised data LUAD_LUSC_data_val_reid.csv stores the test set used for the binary classification. But the LUAD_LUSC_labels_train_reid.csv and the LUAD_LUSC_labels_val_reid.csv do? Why is it a text description? Looking forward to your reply,thanks!

**Comment by ls1rius** [comment]
[fix_pth.yaml](https://github.com/ls1rius/WSI_FiVE/tree/main/configs/wsi/fix_pth.yaml) is used for pre-training, and we do not specifically provide files for few shots, if you want performance few-shot you can sample some data from the whole dataset.

[fix_pth_ft.yaml](https://github.com/ls1rius/WSI_FiVE/tree/main/configs/wsi/fix_pth_ft.yaml) is used for fine-tune.

The difference between two files is that, fix_pth.yaml uses text description of wsi as labels, and fix_pth_ft.yaml uses LUAD/LUSC as labels (binary classification). And the validation set is the same in the two files(binary classification).

Normally, no matter zero-shot, few-shot or supervision experiments are all based on the pre-trained model.

The core of the article is to enhance the model's understanding of semantics through image-text pairs dataset. 
Through fine-grained learning modes to reinforce this learning process.

**Comment by iro-del** [comment]
Thank you very much for your patience! Sorry for bothering you again, I have pre-trained using the features provided by DSMIL based on the trains in the readme, and during the training process the accuracy of TCGA's binary classification of zero shot on LUAD_LUSC_data_val_reid.csv is about 94%, which is much higher than that of the results in Table 3. Is Table 3 in the paper a binary classification of TCGA? How were the results in Table 3 obtained? What are the training and test sets used?
Looking forword to your reply!Thanks!

**Comment by ls1rius** [comment]
The PTH provided by DSMIL has undergone some other pre training, like simclr.
The result of Table 3 combines the results of training method DSMIL and end2end training method.
Similarly, if you conduct subtype experiments, you will find that end2end has higher training performance.
In addition, we have made some tricks to the code, so it will be slightly higher than reported in the paper.
Anyway, Table 3 reports a relatively conservative value that takes into account most of the situations.
